code/scripts/eventhash.js

```diff
@@ -1,33 +1,33 @@
 const fs = require('fs').promises
 const EVENT_LIST = process.env.EVENT_LIST || './scripts/events.txt'
 const EVENT_MAP_OUT = process.env.EVENT_MAP_OUT || './scripts/events-map.txt'
+const EVENT_MAP_OUT_JSON = process.env.EVENT_MAP_OUT_JSON || './lib/events-map.json'
 const ONEUtil = require('../lib/util')
 const PATTERN = /^event ([A-Za-z0-9]+)\((.*)\);$/
 async function main () {
   const list = await fs.readFile(EVENT_LIST, { encoding: 'UTF-8' })
   const lines = list.split('\n')
   const hashMap = {}
+  const hashMapJson = {}
   for (let line of lines) {
     const m = line.match(PATTERN)
     if (!m) {
       continue
     }
     const method = m[1]
     const args = m[2]
-    if (!args) {
```

Comment:
events with empty argument would have `args.split(', ')` to be `['']` and compute the incorrect hash. If this block is removed, `paramTypes` needs to consider this case later

Reply:
About the issue you mentioned. It doesn't end up with two different hash, if it does, you would have `two records with different hash and same event name` in the output(event-map.txt) file, because it is `if statement, not if else statement` the if statement **added into hashMap** and then **continue to run the method to find hash with params and add into the hashMap**.  


''.split(', ') = ['']
[''].join(',') = ''

Reply:
Okay, let me verify and check this later. I can't remember the exact reason I did it like that before, but I recall there are some.

Reply:
yes